contracts/strategies/JoeStrategyV3.sol

```diff
@@ -0,0 +1,318 @@
+            emit Withdraw(msg.sender, depositTokenAmount);
+        }
+    }
+
+    function _withdrawDepositTokens(uint amount) private {
+        require(amount > 0, "JoeStrategyV3::_withdrawDepositTokens");
+        stakingContract.withdraw(PID, amount);
+    }
+
+    function reinvest() external override onlyEOA {
+        uint unclaimedRewards = checkReward();
+        require(unclaimedRewards >= MIN_TOKENS_TO_REINVEST, "JoeStrategyV3::reinvest");
+        (uint poolTokenAmount, address extraRewardTokenAddress, uint extraRewardTokenAmount, uint rewardTokenAmount) = _checkReward();
+        _reinvest(poolTokenAmount, extraRewardTokenAddress, extraRewardTokenAmount, rewardTokenAmount);
+    }
+
+    function _convertRewardIntoWAVAX(uint pendingJoe, address extraRewardToken, uint pendingExtraReward) private returns (uint) {
+        uint convertedAmountWAVAX = 0;
+
+        if (extraRewardToken == address(poolRewardToken)) {
+            convertedAmountWAVAX = DexLibrary.swap(
+                pendingExtraReward.add(pendingJoe),
+                address(poolRewardToken), address(WAVAX),
+                swapPairWAVAXJoe
+            );
+            return convertedAmountWAVAX;
+        }
+
+        convertedAmountWAVAX = DexLibrary.swap(
+            pendingJoe,
+            address(poolRewardToken), address(WAVAX),
+            swapPairWAVAXJoe
+        );
+        if (
+            address(swapPairExtraToken) != address(0)
+            && pendingExtraReward > 0
+            && DexLibrary.checkSwapPairCompatibility(swapPairExtraToken, extraRewardToken, address(WAVAX))
+        ) {
+            convertedAmountWAVAX = convertedAmountWAVAX.add(
+                DexLibrary.swap(pendingExtraReward, extraRewardToken, address(WAVAX), swapPairExtraToken)
+            );
+        }
+        return convertedAmountWAVAX;
+    }
+
+    /**
+     * @notice Reinvest rewards from staking contract to deposit tokens
+     * @dev Reverts if the expected amount of tokens are not returned from `stakingContract`
+     */
+    function _reinvest(uint _pendingJoe, address _extraRewardToken, uint _pendingExtraToken, uint _pendingWavax) private {
+        stakingContract.deposit(PID, 0);
+        uint amount = _pendingWavax.add(
+            _convertRewardIntoWAVAX(_pendingJoe, _extraRewardToken, _pendingExtraToken)
+        );
+
+        uint devFee = amount.mul(DEV_FEE_BIPS).div(BIPS_DIVISOR);
+        if (devFee > 0) {
+            _safeTransfer(address(WAVAX), devAddr, devFee);
+        }
+
+        uint adminFee = amount.mul(ADMIN_FEE_BIPS).div(BIPS_DIVISOR);
+        if (adminFee > 0) {
+            _safeTransfer(address(WAVAX), owner(), adminFee);
+        }
+
+        uint reinvestFee = amount.mul(REINVEST_REWARD_BIPS).div(BIPS_DIVISOR);
+        if (reinvestFee > 0) {
+            _safeTransfer(address(WAVAX), msg.sender, reinvestFee);
+        }
+
+        (uint amountOutToken0, uint amountOutToken1) = DexLibrary.convertWAVAXToDepositTokens(
```

Comment:
And here would work too, we would just need to make sure we pass the argument rewardToken as the WAVAX defined in the contract